docs/adding_objectives.rst

```diff
@@ -129,6 +137,10 @@ A full example objective with comments describing key points is given below:
             # equilibrium, though in most cases that isn't necessary.
             self._profiles = get_profiles(self._data_keys, eq=eq, grid=self.grid)
             self._transforms = get_transforms(self._data_keys, eq=eq, grid=self.grid)
```

Comment:
this is not specific to this PR and I should probably just make an issue, but these lines using the  helper functions are a bit opaque as it is not clear what self._transforms etc are, I guess is not that important but just thought for clarity could be good to have this be more transparent of at least what type of object are and what they contain